dask/dataframe/shuffle.py

```diff
@@ -174,17 +176,17 @@ def sort_values(
 
 
 def set_index(
-    df,
-    index,
-    npartitions=None,
-    shuffle=None,
-    compute=False,
-    drop=True,
-    upsample=1.0,
-    divisions=None,
-    partition_size=128e6,
+    df: DataFrame,
+    index: Union[str, Series],
+    npartitions: Optional[Union[int, Literal["auto"]]] = None,
+    shuffle: Optional[str] = None,
+    compute: bool = False,
+    drop: bool = True,
+    upsample: float = 1.0,
+    divisions: Optional[Sequence] = None,
```

Comment:
@crusaderky I realized, rereading your comment, that using `Sequence` to mean `Sequence[Any]` is preferred, to avoid a second import. Is that correct? If so, @jorloplaz, you can ignore all the comments about `Sequence`.

Reply:
Yes, when I don't type it's the same as typing with `[Any]`, with the advantage that I don't need the double import. So many of the things are related to divisions, and since we can have anything in the index, those are always `Any`, so that's why I used plain `List` or `Tuple`. It's true though that the `lens` should be typed to `int`.